### DISCUSSION TITLE
Different auth for the same host but different ports via `hostRules`

### DISCUSSION THREAD

**Discussion by catostrophe** [renovatebot/renovate#17664] [Request Help]
I use a private Nexus as a docker registry.

My own images are hosted at `nexus.mydomain.com`. Dockerhub is proxied through `nexus.mydomain.com:7777` (another port). 
I refer to private images as `nexus.mydomain.com/group/image:tag`, to public images at Dockerhub as `nexus.mydomain.com:7777/group/image:tag`.

The proxy to dockerhub is anonymous, the private registry requires auth.

How should I set up `hostRules` to make it work?

I tried this:
```json
     [
        {
          "hostType": "docker",
          "matchHost": "nexus.mydomain.com",
          "username": "$NEXUS_USER",
          "password": "$NEXUS_PASS"
        }
      ]
```
but it breaks the proxy.

I also tried this:
```json
     [
        {
          "hostType": "docker",
          "matchHost": "https://nexus.mydomain.com/",
          "username": "$NEXUS_USER",
          "password": "$NEXUS_PASS"
        }
      ]
```
but it doesn't work, probably because I don't use `https://` in image names.

How can I solve this problem?

**Comment by rarkins** [comment]
`matchHost` with `https://` prefix should work, even if the image name is without the `https://`. Please retry it and provide more details about "it doesn't work"

**Reply by catostrophe** [reply]
I get: "Failed to get authHeaders for getTags lookup" and "Failed to look up dependency nexus.mydomain.com/group/image"

**Reply by rarkins** [reply]
This doesn't necessarily mean that the `hostRules` aren't working.

I think you should have one rule with `"matchHost": "https://nexus.mydomain.com/",` and another with `"matchHost": "https://nexus.mydomain.com:7777/",`.

**Reply by catostrophe** [reply]
And what should be "username", "password" for the public registry?

**Reply by catostrophe** [reply]
This seems to work:
```json
      [
        {
          "hostType": "docker",
          "matchHost": "https://nexus.mydomain.com/",
          "username": "NEXUS_USER",
          "password": "$NEXUS_PASS"
        },
        {
          "hostType": "docker",
          "matchHost": "https://nexus.mydomain.com:7777/"
        }
      ]
```

**Reply by rarkins** [reply]
In that case the second rule actually does nothing, so you could remove it
